### ISSUE TITLE
Consuming SQS messages from other sources

### ISSUE THREAD

**Issue by bwegrzyn** [php-enqueue/enqueue-dev#296]
I currently am using AWS SES -> SNS -> SQS to process incoming emails for my app. The whole SES -> SNS -> SQS process works fine, and SNS creates a message in SQS, but that is where the problems start.

I've set queueName to my SQS queue name and queueNameHardcoded to true, but when the consumer tries to process the message, it gets the following error:

```php
class IncomingEmailProcessor implements PsrProcessor, TopicSubscriberInterface
{

    private $logger;

    public function __construct(LoggerInterface $logger)
    {
        $this->logger = $logger;
    }

    public static function getSubscribedTopics()
    {
        return [
            self::class => [
                'queueName' => 'IncomingEmail',
                'queueNameHardcoded' => true,
                //'processorName' => self::class // Doesn't work when uncommented either
            ]
        ];
    }

    public function process(PsrMessage $message, PsrContext $session)
    {
        $this->logger->info('Got message', [
            'message' => $message->getBody()
        ]);

        return self::ACK;
    }


}
```

```
In DelegateProcessor.php line 31:
                                                                    
  [LogicException]                                                  
  Got message without required parameter: "enqueue.processor_name"  
                                                                    

Exception trace:
 Enqueue\Client\DelegateProcessor->process() at /app/vendor/enqueue/enqueue/Consumption/QueueConsumer.php:292
 Enqueue\Consumption\QueueConsumer->doConsume() at /app/vendor/enqueue/enqueue/Consumption/QueueConsumer.php:221
 Enqueue\Consumption\QueueConsumer->consume() at /app/vendor/enqueue/enqueue/Symfony/Client/ConsumeMessagesCommand.php:115
 Enqueue\Symfony\Client\ConsumeMessagesCommand->execute() at /app/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at /app/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:954
 Symfony\Component\Console\Application->doRunCommand() at /app/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:83
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /app/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:240
 Symfony\Component\Console\Application->doRun() at /app/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:71
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /app/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:148
 Symfony\Component\Console\Application->run() at /app/bin/console:29
```

Is it not possible to consume messages from sources other than Enqueue?

This issue seems similar to #196, which was had tests added and then closed as invalid.

**Comment by makasim** [comment]
The options `queueName` and `queueNameHardcoded` work only for commands now. Events are not supported.  

Though it most likely could be implemented.

**Comment by bwegrzyn** [comment]
So there is no way to currently process queue messages from sources other the enqueue library?

**Comment by makasim** [comment]
events - no, commands - yes

**Comment by bwegrzyn** [comment]
Would you be able to provide some details on how to do it with commands? I've looked at the docs but I can't find much about it there and I'm not sure what the differences between commands and events are.

**Comment by makasim** [comment]
Create a method like this https://github.com/php-enqueue/enqueue-elastica-bundle/blob/master/Queue/PopulateProcessor.php#L127

and remove getSubscribedTopics method.

**Comment by makasim** [comment]
@bwegrzyn have you been able to solve the issue?

**Comment by bwegrzyn** [comment]
Well... I changed my processor class to use CommandSubscriberInterface and QueueSubscriberInterface like in the example, and it seemed like it was working. It was receiving messages correctly. 

Then I was making some changes and I seem to have broken it again. It fails with the exact same error as before. I have no idea why it isn't working now.

```php
class IncomingEmailProcessor implements PsrProcessor, CommandSubscriberInterface, QueueSubscriberInterface
{
    private $logger;

    public function __construct(LoggerInterface $logger)
    {
        $this->logger = $logger;
    }

    public static function getSubscribedCommand()
    {
        return [
            'processorName' => self::class,
            'queueName' => 'IncomingEmail',
            'queueNameHardcoded' => true
        ];
    }

    public static function getSubscribedQueues()
    {
        return [self::class];
    }

    public function process(PsrMessage $message, PsrContext $session)
    {
        $this->logger->info('Got Message', [
            'message' => $message->getBody()
        ]);

        return self::ACK;
    }
}
```

**Comment by bwegrzyn** [comment]
I think I figured it out... I was confused about the `exclusive` property. I thought it meant that this command can only be processed one at a time, so I had removed it after it was working, but I had my messages set to requeue for testing purposes.

After I removed the `exclusive` property, I'm guessing it was still getting the message properly because the `ExclusiveCommandExtension` had added it previously. 

Anyways... the `exclusive` property doesn't mean what I thought it meant so it looks like its working now.
